extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/RestClientMetricsFilter.java

```diff
@@ -21,6 +21,7 @@
  */
 @Unremovable
 @Provider
+@SuppressWarnings("unused") // this is used by io.quarkus.micrometer.deployment.binder.HttpBinderProcessor
```

Comment:
I'm surprised that you get unused warnings for public classes? I think they are usually considered public APIs and thus not marked as unused.